package exception;

public class ChainedTest {

	public static void main(String[] args) {
		install();
		
//		Throwable initCause (Throwable cause) // 지정한 예외를 원인 예외로 등록
//		Throwable getCause() // 원인 예외를 반환
		
		
		
	}

	static void install() {
		 try {
			startInstall();
		} catch (SpaceException e) {
			System.out.println("ERROR : " + e.getMessage());
		}
//		 copyFiles();
		 
	}


	static void startInstall() throws SpaceException {
		System.out.println("설치를 시작합니다.");
		if(!enoughSpace()) {
			throw new SpaceException ("디스크 공간이 부족합니다.");
		};
	}

	static boolean enoughSpace() {

		return false;
	}

	private static void copyFiles() {
		System.out.println("파일을 복사합니다.");		
	}
	
}


class SpaceException extends Exception {
	// 사용자 정의 예외 클래스
	SpaceException(String msg) {
		super(msg);
	}
}